packages/pretty-format/src/plugins/DOMCollection.ts

```diff
@@ -47,8 +44,14 @@ export const serialize: NewPlugin['serialize'] = (
     (OBJECT_NAMES.indexOf(name) !== -1
       ? '{' +
         printObjectProperties(
-          name === 'NamedNodeMap'
-            ? Array.prototype.reduce.call(collection, propsReducer, {})
+          isNamedNodeMap(collection)
+            ? Array.from(collection).reduce<Record<string, string>>(
```

Comment:
I inlined this function as there's no way to pass a generic parameter otherwise; it's inlined in `DOMElement`, and so now actually matches 1:1.